rocketpy/tools.py

```diff
@@ -0,0 +1,21 @@
+_NOT_FOUND = object()
+
+
+class cached_property:
+    def __init__(self, func):
+        self.func = func
+        self.attrname = None
+        self.__doc__ = func.__doc__
+
+    def __set_name__(self, owner, name):
+        self.attrname = name
+
+    def __get__(self, instance, owner=None):
+        if instance is None:
+            return self
+        cache = instance.__dict__
```

Comment:
I prefer keeping the exception here as in `functools.cached_property` source code, since it prevents the decorator being used on a method that is called _before_ `__set_name__`.

```
if self.attrname is None:
            raise TypeError(     
    "Cannot use cached_property instance without calling __set_name__ on it.")
```

Reply:
I have accepted and committed your suggestion @phmbressan. I have also requested another final review.

Out of curiosity, can you share an example of when such a thing can happen? The decorator being used on a method that is called before __set_name__? Would it be like trying to use the decorator in __init__?

Reply:
I have thought a bit and I don't think it is quite possible to make a method that is called before `__set_name__` without too cumbersome implementations. I believe that what may trigger the exception is bypassing the `__set_name__` method as shown [here](https://stackoverflow.com/questions/62160411/pythons-new-functools-cached-property-bug-or-limitation).